File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/javadoc/comment/direct/Java/Go/s185737640_commented.go
package main

import (
	"fmt"
)

func main() {
	var S string
	fmt.Scanln(&S)

	if len(S)%2 == 1 {
		S = S[:len(S)-1]
	} else {
		S = S[:len(S)-2]
	}

	for len(S) > 0 {
		if isEvenString(S) {
			break
		}
		S = S[:len(S)-2]
	}
	fmt.Println(len(S))
}

func isEvenString(s string) bool {
	a := s[:len(s)/2]
	b := s[len(s)/2:]
	return a == b
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Python/s185737640_commented.py
def is_even_string(s):
    return s[:len(s)//2] == s[len(s)//2:]

s = input().strip()

# Ensure the length of s is even
if len(s) % 2 == 1:
    s = s[:-1]

# Further ensure the length of s is at least 2
if len(s) > 1:
    s = s[:-2]

while len(s) > 0:
    if is_even_string(s):
        break
    s = s[:-2]

print(len(s))  # End of Code
